### PR TITLE
Handlebars reloadable templates

### PR DESCRIPTION
**Pull request by qnkhuat**
this makes handlebars reload resource if it's changed. Good for development purposes.

UPDATE: apparently even with my custom implementation the test is still failing on some java versions, so I've reverted back to use `ConcurrentMapTemplateCache`. It works in dev: if you update the content of the file, the cache is invalidated right away. So I guess it was just a test setup issue.

~~I had to write a custom cacher because there is a bug in the ConcurrentMapTemplateCache implementation~~

~~where this test will fail~~
```clojure
(deftest reload-template-if-it's-changed
  (testing "reload template if it's changed"
    (with-temp-template! [tmpl-name (u.random/random-name) "Hello {{name}}"]
      (is (= "Hello Ngoc" (handlebars/render tmpl-name {:name "Ngoc"})))
      (spit (format "test_resources/%s" tmpl-name) "Hello {{name}} updated!")
      (is (= "Hello Ngoc updated!" (handlebars/render tmpl-name {:name "Ngoc"}))))))
```

~~but if I add this one line, it'll pass~~

```diff
(deftest reload-template-if-it's-changed
  (testing "reload template if it's changed"
    (with-temp-template! [tmpl-name (u.random/random-name) "Hello {{name}}"]
      (is (= "Hello Ngoc" (handlebars/render tmpl-name {:name "Ngoc"})))
+     (handlebars/render tmpl-name {:name "Ngoc"})
      (spit (format "test_resources/%s" tmpl-name) "Hello {{name}} updated!")
      (is (= "Hello Ngoc updated!" (handlebars/render tmpl-name {:name "Ngoc"}))))))
```

### CODE REVIEW

**Comment by qnkhuat** [comment]
oh god, that test is failing on CI but no locally 🤔

File: .github/workflows/backend.yml [hunk 1]
```diff
@@ -168,7 +168,7 @@ jobs:
 
       - name: Run tests
         id: run-java-tests
-        run: clojure -X:dev:ci:test:${{ matrix.edition }}:${{ matrix.edition }}-dev
+        run: clojure -X:dev:ci:test:${{ matrix.edition }}:${{ matrix.edition }}-dev :only metabase.channel.template.handlebars-test
```
Comment (crisptrutski) [inline .github/workflows/backend.yml:171]: For reproducing on CI faster? Never thought of this!

**Comment by qnkhuat** [comment]
hmm, it's super weird but one thing for sure that the default implementation fails on all of java version: https://github.com/metabase/metabase/actions/runs/11832526701/job/32969513565

where my implementation only fail on java-11-oss and java-21-ee : https://github.com/metabase/metabase/actions/runs/11892311974/job/33134912503

**Comment by qnkhuat** [comment]
update: I've removed the test as it's still [failing](https://github.com/metabase/metabase/actions/runs/11905934912/job/33177217758) even with the trick. This is just for dev-convenience so I think it's fine

File: src/metabase/channel/template/handlebars.clj [hunk 2]
```diff
@@ -14,25 +14,27 @@
...
-
 (defn- wrap-context
   [context]
   (walk/postwalk
    #(if (keyword? %)
       (u/qualified-name %)
       %) context))
 
+(def ^:private default-hbs
+  (registry (classpath-loader "/" "") :reload? true))
```
[the remark this answers is not in the document]
Comment (crisptrutski) [reply src/metabase/channel/template/handlebars.clj:36]: Doesn't really matter in practice, but communicates the intent, and protects against this becoming a function etc

**Review by crisptrutski — APPROVED** [review]

**Pull request merged**